cogs/heartbeat/testCommandsCore.py

```diff
@@ -0,0 +1,99 @@
+import asyncio
+from typing import Union
+from unittest import mock
+
+import pytest
+
+from redbot.core.bot import Red
+from redbot.core.commands.context import Context
+
+from .heartbeat import Heartbeat
+from .constants import KEY_INSTANCE_NAME, KEY_INTERVAL, KEY_PUSH_URL, MIN_INTERVAL
+
+
+@pytest.mark.asyncio
+async def testCmdCheck(red: Red):
+    """Test to ensure `cmdCheck` works as expected."""
+
+    # prep
+    red.loop = mock.create_autospec(spec=asyncio.AbstractEventLoop)
+    cog = Heartbeat(bot=red)
+    ctx: Union[mock.Mock, Context] = mock.create_autospec(spec=Context)
+    ctxSend: mock.Mock = ctx.send
+    # test
+    instanceName = "test instance"
+    await cog.cmdName(ctx=ctx, name=instanceName)
+    await cog.cmdCheck(ctx=ctx)
+    expectedReply = f"**{instanceName}** is responding."
+    actualReply = ctxSend.call_args.kwargs.get("content") or ctxSend.call_args.args[0]
+    assert actualReply == expectedReply
+
+
+@pytest.mark.asyncio
+async def testCmdUrl(red: Red):
+    """Test to ensure `cmdUrl` works as expected."""
+
+    red.loop = mock.create_autospec(spec=asyncio.AbstractEventLoop)
+    cog = Heartbeat(bot=red)
+    ctx: Union[mock.Mock, Context] = mock.create_autospec(spec=Context)
```

Comment:
I did think about this point briefly. We can certainly have a fixture that denotes _a fake Red that has cog Heartbeat loaded_ and another fixture for _a fake `Context` object_.

Reply:
Since `cog` is called in subsequent lines, do we want to do something like `red.get_cog(Heartbeat.__name__)`? Looks clunky to me.

Reply:
Oh actually, we just test the cog itself and not Red, so the fixture can be _the cog that is loaded by Red_ instead of _Red that has the cog loaded_. 😅